fpga.chart.py

```diff
@@ -177,12 +191,20 @@ def _get_fpga_power(self, idx):
             return self._get_xilinx_fpga_power(idx)
 
 
+    def get_fpga_os_status(self):
```

Comment:
Overall semantics: the method name invites to call it directly, which is not intended anymore. I would propose to have a run-method which starts the thread and implements the get-method as well. Like:

    def run_thread(self):
        def get_fpga_os_status(x):
            [collection logic]
            x.temp = 123

        [start_thread_logic](target=get_fpga_os_status, args=(self, ))